src/Model.php

```diff
@@ -763,4 +795,140 @@ function ($relation)
 			)
 		);
 	}
+
+	/**
+	 * Get the columns from database table.
+	 *
+	 * @param   boolean         $reload  flag to reload cache
+	 *
+	 * @return  mixed  An array of the field names, or false if an error occurs.
+	 *
+	 * @throws  \UnexpectedValueException
+	 */
+	public function getDefaultFields($reload = false)
+	{
+		// Lookup the fields for this table only once.
+		if (!isset(static::$fieldsCache[$this->getTableName()]) || $reload)
+		{
+			$fields = $this->db->getTableColumns($this->getTableName());
+
+			if (empty($fields))
+			{
+				throw new \UnexpectedValueException(sprintf('No columns found for %s table', $this->getTableName()));
+			}
+
+			$fields = array_map(
+				function ($field)
+				{
+					return null;
+				},
+				$fields
+			);
+
+			static::$fieldsCache[$this->getTableName()] = array_keys($fields);
+		}
+
+		return static::$fieldsCache[$this->getTableName()];
+	}
+
+	/**
+	 * Reset function
+	 * Will not throw an error if the column does not exist during reset, but it will be thrown when saving the model.
+	 *
+	 * @param   array          $attributes  pre loads any attributed for the model (user friendly format)
+	 *
+	 * @return void
+	 */
+	public function reset(array $attributes = [])
```

Comment:
All these methods from here to the bottom do not belong in the framework - they belong in the cms.